dot/network/discovery.go

```diff
@@ -49,9 +54,16 @@ type discovery struct {
 	ds                 *badger.Datastore
 	pid                protocol.ID
 	minPeers, maxPeers int
+
+	metrics map[string]interface{}
 }
 
 func newDiscovery(ctx context.Context, h libp2phost.Host, bootnodes []peer.AddrInfo, ds *badger.Datastore, pid protocol.ID, min, max int) *discovery {
+	metrics.Enabled = true
```

Comment:
seems odd that `metrics` gets enabled in this constructor.  Why isn't `metrics` an attribute of `disovery`?